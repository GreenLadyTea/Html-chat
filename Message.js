import React from 'react';
class Message extends React.Component {
    render()
    {
        const{nick, message} = this.props;
        return <div>
            <b>{nick}: </b>
            {message}
        </div>;
    }
}
export default Message;