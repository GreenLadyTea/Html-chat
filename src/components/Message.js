import React from "react";

export default class Message extends React.Component {
  render() {
    const { nickname, content } = this.props;
    return (
      <div>
        <b>{nickname}: </b>
        {content}
      </div>
    );
  }
}
