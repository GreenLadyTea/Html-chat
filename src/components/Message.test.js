import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Message from "./Message";
import { describe, test } from "@jest/globals";

describe("Message component", () => {
  test("Message matches snapshot", () => {
    const nickname = "test";
    const content = "test";
    const component = renderer.create(<Message nickname={nickname} content={content} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("Message shows nick and message", () => {
    const nickname = "test";
    const content = "test";
    const component = shallow(<Message nickname={nickname} content={content} />);
    expect(component.text()).toContain(nickname);
    expect(component.text()).toContain(content);
  });
});
