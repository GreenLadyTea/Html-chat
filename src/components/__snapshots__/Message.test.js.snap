// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component Message matches snapshot 1`] = `
<div>
  <b>
    test
    : 
  </b>
  test
</div>
`;
